### PR TITLE
adding test suite

### PR DESCRIPTION
**Pull request by tinakurian**
Adding test suite

### CODE REVIEW

File: test/testsuite.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	"os"
+
+ 	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+	"github.com/stretchr/testify/suite"
+)
+
+ // UnitTestSuite is the base test suite for unit tests.
+type UnitTestSuite struct {
+	suite.Suite
+	ConfigRegistry *configuration.Registry
```
Comment (alexeykazakov) [inline test/testsuite.go]: Can we call it just `Config`?
```suggestion
	Config *configuration.Registry
```

File: pkg/controller/authconfig_test.go [hunk 2]
```diff
@@ -10,30 +10,40 @@ import (
...
 	req, err := http.NewRequest(http.MethodGet, "/api/v1/authconfig", nil)
-	require.NoError(t, err)
+	require.NoError(s.T(), err)
 
 	// Create logger and registry.
 	logger := log.New(os.Stderr, "", 0)
 	configRegistry := configuration.CreateEmptyRegistry()
 
 	// Set the config for testing mode, the handler may use this.
 	configRegistry.GetViperInstance().Set("testingmode", true)
```
Comment (alexeykazakov) [inline pkg/controller/authconfig_test.go]: You don't need to create a new CreatEmptyRegistry and set it to the testing mode now. Use `s.Config` where you need the configuration. It's already set to the testing mode for the suite. That was the main reason for introducing the suite: https://github.com/codeready-toolchain/registration-service/pull/29/files#diff-5626d2d6a4ce3579865081a105c86127R19-R27

File: pkg/controller/signup_test.go [hunk 3]
```diff
@@ -9,30 +9,40 @@ import (
...
 	req, err := http.NewRequest(http.MethodPost, "/api/v1/signup", nil)
-	require.NoError(t, err)
+	require.NoError(s.T(), err)
 
 	// Create logger and registry.
 	logger := log.New(os.Stderr, "", 0)
 	configRegistry := configuration.CreateEmptyRegistry()
 
 	// Set the config for testing mode, the handler may use this.
 	configRegistry.GetViperInstance().Set("testingmode", true)
```
Comment (alexeykazakov) [inline pkg/controller/signup_test.go]: Same here.

File: pkg/controller/health_check_test.go [hunk 4]
```diff
@@ -10,30 +10,39 @@ import (
...
 	req, err := http.NewRequest(http.MethodGet, "/api/v1/health", nil)
-	require.NoError(t, err)
+	require.NoError(s.T(), err)
 
 	// Create logger and registry.
 	logger := log.New(os.Stderr, "", 0)
 	configRegistry := configuration.CreateEmptyRegistry()
 
 	// Set the config for testing mode, the handler may use this.
 	configRegistry.GetViperInstance().Set("testingmode", true)
```
Comment (alexeykazakov) [inline pkg/controller/health_check_test.go]: Same here.

File: pkg/controller/authconfig_test.go (hunk 2 above)
Comment (tinakurian) [reply pkg/controller/authconfig_test.go]: fixed here: 6f5b359318a99b71809fc28cc97837c6a906413b

File: pkg/controller/health_check_test.go (hunk 4 above)
Comment (tinakurian) [reply pkg/controller/health_check_test.go]: fixed here: 6f5b359318a99b71809fc28cc97837c6a906413b

File: pkg/controller/signup_test.go (hunk 3 above)
Comment (tinakurian) [reply pkg/controller/signup_test.go]: fixed here: 6f5b359318a99b71809fc28cc97837c6a906413b

File: test/testsuite.go (hunk 1 above)
Comment (tinakurian) [reply test/testsuite.go]: yup! fixed here: 1b2cb31ed23212e813dc3dee0fa909e4a0d63551

**Pull request merged**